calico_containers/tests/st/workload.py

```diff
@@ -0,0 +1,100 @@
+from netaddr import IPAddress, AddrFormatError
+
+
+class Workload(object):
+    """
+    A calico workload.
+
+    These are the end-users containers that will run application-level software.
+    """
+    def __init__(self, host, name, ip=None, image="busybox", use_powerstrip=True):
+        """
+        Create the workload and detect its IPs.
+
+        :param host: The host container on which this workload is instantiated.
+        All commands executed by this container will be passed through the host
+        via docker exec.
+        :param name: The name given to the workload container. This name is
+        passed to docker and can be used inside docker commands.
+        :param ip: The IP to be assigned to this workload via calico. May be
+        either IPv4 or IPv6. Calico supports multiple IPs per workload, but
+        this testing framework does not yet.
+        :param image: The docker image to be used to instantiate this
+        container. busybox used by default because it is extremely small and
+        has ping.
+        :param use_powerstrip: Use Powerstrip in front of docker to inform
+        calico of networking changes.
+        """
+        self.host = host
+        self.name = name
+
+        args = [
+            "docker", "run",
+            "--tty",
+            "--interactive",
+            "--detach",
+            "--name", name,
+        ]
+        if ip:
+            # Powerstrip passes the CALICO_IP var to calico.
+            args += ["--env", "CALICO_IP=%s" % ip]
+        args.append(image)
+        command = ' '.join(args)
+
+        host.execute(command, use_powerstrip=use_powerstrip)
+
+        # There is an unofficial ip=auto option in addition to ip=None.
+        if ip is None or ip == 'auto':
+            version = None
+        else:
+            version = IPAddress(ip).version
+
+        # if version == 6:
```

Comment:
If you're going to leave this commented out code, then you need a comment explaining why it's being worked around and what has to happen before we can remove the workaround.